### PR TITLE
Feature/fix local envionment

### PR DESCRIPTION
**Pull request by caiola**

### CODE REVIEW

File: backend/api/models/ads.py [hunk 1]
```diff
@@ -13,10 +14,11 @@ class Ad(db.Model, BaseModel, metaclass=MetaBaseModel):
 
     __tablename__ = "ads"
 
-    id = db.Column(db.BigInteger(), primary_key=True)
-    uuid = db.Column(db.String(50), default=uuid.uuid4, unique=True)
-    title = db.Column(db.String(100), nullable=False, unique=True)
+    id = db.Column(db.BigInteger(), primary_key=True, nullable=False)
```
Comment (sergiodenboer) [inline backend/api/models/ads.py:17]: These changes need migrations

File: backend/api/models/users.py [hunk 2]
```diff
@@ -10,6 +10,6 @@ class User(db.Model, BaseModel, metaclass=MetaBaseModel):
 
     __tablename__ = "users"
 
-    id = db.Column(db.BigInteger(), primary_key=True)
+    id = db.Column(db.BigInteger(), primary_key=True, nullable=False)
```
Comment (sergiodenboer) [inline backend/api/models/users.py:13]: These changes need migrations.

File: backend/api/models/ads.py [hunk 3]
```diff
@@ -13,10 +14,11 @@ class Ad(db.Model, BaseModel, metaclass=MetaBaseModel):
...
+    uuid = db.Column(db.String(50), default=uuid.uuid4, unique=True, nullable=False)
```
Comment (sergiodenboer) [inline backend/api/models/ads.py:18]: Can we change the varchar [size to 36](https://stackoverflow.com/questions/13397038/uuid-max-character-length)?

File: backend/api/models/ads.py [hunk 4]
```diff
@@ -13,10 +14,11 @@ class Ad(db.Model, BaseModel, metaclass=MetaBaseModel):
...
+    title = db.Column(db.String(100), nullable=False)
     description = db.Column(db.Text(), nullable=False)
-    user_id = db.Column(db.BigInteger, db.ForeignKey("users.id"), nullable=False)
+    user_id = Column(db.BigInteger(), ForeignKey('users.id', name="ads_user_id_fk_users_id", onupdate="RESTRICT",
+                     ondelete="RESTRICT"), nullable=False)
```
Comment (sergiodenboer) [inline backend/api/models/ads.py:22]: I would be Ok with cascading but I understand the drawback.

File: backend/api/models/ads.py [hunk 5]
```diff
@@ -13,10 +14,11 @@ class Ad(db.Model, BaseModel, metaclass=MetaBaseModel):
 
     __tablename__ = "ads"
 
-    id = db.Column(db.BigInteger(), primary_key=True)
-    uuid = db.Column(db.String(50), default=uuid.uuid4, unique=True)
-    title = db.Column(db.String(100), nullable=False, unique=True)
+    id = db.Column(db.BigInteger(), primary_key=True, nullable=False)
+    uuid = db.Column(db.String(50), default=uuid.uuid4, unique=True, nullable=False)
+    title = db.Column(db.String(100), nullable=False)
```
Comment (sergiodenboer) [inline backend/api/models/ads.py:19]: Cna we add tje constraint you've suggest in a previous MR? title and user must be unique?

File: .env.local.sample [hunk 6]
```diff
@@ -16,18 +14,18 @@ FLASK_ENV=DEV
...
 APPLICATION_MYSQL_HOST=dbserver
 APPLICATION_MYSQL_PORT=3306
-APPLICATION_MYSQL_DB=db
+APPLICATION_MYSQL_DATABASE=db
 APPLICATION_MYSQL_PREFIX=dev_
 
 TIMEZONE=Europe/Lisbon
 
 # We could use python:3.7.16-slim but have to deal with gcc and mysqclient dependencies
 PYTHON_VERSION=3.10
```
Comment (sergiodenboer) [inline .env.local.sample:26]: This is not needed has the version is controlled via package manager and Docker Images.

File: backend/api/config.py [hunk 7]
```diff
@@ -12,11 +12,11 @@
 SQLALCHEMY_TRACK_MODIFICATIONS = False
 
 MYSQL = {
-    "user": os.environ["MYSQL_USER"],
-    "pw": os.environ["MYSQL_PASSWORD"],
-    "host": os.environ["MYSQL_HOST"],
-    "port": os.environ["MYSQL_PORT"],
-    "db": os.environ["MYSQL_DATABASE"],
+    "user": os.environ["APPLICATION_MYSQL_USER"],
```
Comment (sergiodenboer) [inline backend/api/config.py:15]: Don't really se the need for the `APPLICATION` prefix. If we were to have a read replica the name would become unnecessarily big?

File: services/flake/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9-slim
```
Comment (sergiodenboer) [inline services/flake/Dockerfile:1]: Having docker file for this is something we should not do.
There are better ways. We would primarly run this IN CI and local development. This is more a code quality concern than an orchestration concern.

File: backend/api/models/ads.py (hunk 4 above)
Comment (caiola) [reply backend/api/models/ads.py:22]: The default on the default after running the migrations was RESTRICT. I added the value on source code to be explicit

File: backend/api/models/ads.py (hunk 5 above)
Comment (caiola) [reply backend/api/models/ads.py:19]: done

File: backend/api/models/ads.py (hunk 1 above)
Comment (caiola) [reply backend/api/models/ads.py:17]: done, improved naming convention

File: services/flake/Dockerfile (hunk 8 above)
Comment (caiola) [reply services/flake/Dockerfile:1]: yeah, it is not working either way. I will delete it or move it to a folder /tools/

**Pull request merged**